nifi-nar-bundles/nifi-media-bundle/nifi-media-processors/src/test/java/org/apache/nifi/processors/media/TestExtractMediaMetadata.java

```diff
@@ -0,0 +1,411 @@
+        runner.run();
+
+        runner.assertAllFlowFilesTransferred(ExtractMediaMetadata.SUCCESS, 1);
+        runner.assertTransferCount(ExtractMediaMetadata.FAILURE, 0);
+
+        final List<MockFlowFile> successFiles = runner.getFlowFilesForRelationship(ExtractMediaMetadata.SUCCESS);
+        MockFlowFile flowFile0 = successFiles.get(0);
+        flowFile0.assertAttributeExists("filename");
+        flowFile0.assertAttributeEquals("filename", "notImage.txt");
+        flowFile0.assertAttributeNotExists("txt.Content-Type");
+        flowFile0.assertAttributeNotExists("txt.X-Parsed-By");
+    }
+
+    @Test
+    public void testMimeTypeFilter() throws IOException {
+        final TestRunner runner = TestRunners.newTestRunner(new ExtractMediaMetadata());
+        runner.setProperty(ExtractMediaMetadata.FILENAME_FILTER, ".*[tTxXtT]");
+        runner.setProperty(ExtractMediaMetadata.MIME_TYPE_FILTER, "doc.*");
+        runner.setProperty(ExtractMediaMetadata.METADATA_KEY_PREFIX, "txt.");
+        runner.assertValid();
+
+        runner.enqueue(new File("target/test-classes/notImage.txt").toPath());
+        runner.run();
+
+        runner.assertAllFlowFilesTransferred(ExtractMediaMetadata.SUCCESS, 1);
+        runner.assertTransferCount(ExtractMediaMetadata.FAILURE, 0);
+
+        final List<MockFlowFile> successFiles = runner.getFlowFilesForRelationship(ExtractMediaMetadata.SUCCESS);
+        MockFlowFile flowFile0 = successFiles.get(0);
+        flowFile0.assertAttributeExists("filename");
+        flowFile0.assertAttributeEquals("filename", "notImage.txt");
+        flowFile0.assertAttributeNotExists("txt.Content-Type");
+        flowFile0.assertAttributeNotExists("txt.X-Parsed-By");
+    }
+
+    @Test
+    public void testMetadataKeyFilter() throws IOException {
+        final TestRunner runner = TestRunners.newTestRunner(new ExtractMediaMetadata());
+        runner.setProperty(ExtractMediaMetadata.METADATA_KEY_FILTER, "(X-Parsed.*)");
+        runner.setProperty(ExtractMediaMetadata.METADATA_KEY_PREFIX, "txt.");
+        runner.assertValid();
+
+        runner.enqueue(new File("target/test-classes/notImage.txt").toPath());
+        runner.run();
+
+        runner.assertAllFlowFilesTransferred(ExtractMediaMetadata.SUCCESS, 1);
+        runner.assertTransferCount(ExtractMediaMetadata.FAILURE, 0);
+
+        final List<MockFlowFile> successFiles = runner.getFlowFilesForRelationship(ExtractMediaMetadata.SUCCESS);
+        MockFlowFile flowFile0 = successFiles.get(0);
+        flowFile0.assertAttributeExists("filename");
+        flowFile0.assertAttributeEquals("filename", "notImage.txt");
+        flowFile0.assertAttributeExists("txt.X-Parsed-By");
+        assertTrue(flowFile0.getAttribute("txt.X-Parsed-By").contains("org.apache.tika.parser.DefaultParser"));
+        assertTrue(flowFile0.getAttribute("txt.X-Parsed-By").contains("org.apache.tika.parser.txt.TXTParser"));
```

Comment:
This should assert that the attributes that were filtered out are not added to the FlowFile